### PR TITLE
CHE-10657 add support for the macros 'current.project.path'

### PR DESCRIPTION
**Pull request by olexii4**
Add support for the macros 'current.project.path'.

Related issue: https://github.com/eclipse/che/issues/10657

Signed-off-by: Oleksii Orel <oorel@redhat.com>

### CODE REVIEW

**Comment by azatsarynnyy** [comment]
@olexii4 please consider moving the `current.project.path` variable contribution to the separate class, e.g. `ProjectPathVariableContribution` and leaving the `ServerVariablesContribution` as it is. When we have more such variables, it will be harder to support all of them within the one class.

**Comment by olexii4** [comment]
Done

**Comment by azatsarynnyy** [comment]
Thank you!

**Review by azatsarynnyy — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Review by evidolob — APPROVED** [review]

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    async registerVariables(variables: VariableRegistry): Promise<void> {
+
+        variables.registerVariable({
+            name: 'current.project.path',
+            description: 'The path of the project root folder',
+            resolve: async () => {
+                const wsRoot = await this.workspaceService.root;
+                if (!wsRoot || !wsRoot.uri) {
+                    return undefined;
```
Comment (sunix) [inline che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: for all those `return undefined`, do we want to give more feedback to the user on the reason why it couldn't resolve a variable ? @slemeur ? rather than silently exiting

Comment (olexii4) [reply che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: I will add error notifications

**Comment by olexii4** [comment]
done
selection_015
selection_014
...

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts [hunk 2]
```diff
@@ -28,33 +29,47 @@ export class ProjectPathVariableContribution implements VariableContribution {
...
             name: 'current.project.path',
             description: 'The path of the project root folder',
             resolve: async () => {
-                const wsRoot = await this.workspaceService.root;
-                if (!wsRoot || !wsRoot.uri) {
+                let errorMessage = 'Get \'current.project.path\' variable error. ';
+
+                const selection = this.selectionService.selection;
+                if (!Array.isArray(selection) || !selection[0] || !selection[0].fileStat || !selection[0].fileStat.uri) {
+                    errorMessage += 'Any selections are not defined in the project tree.';
```
Comment (sunix) [inline che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: maybe rephrase this with
`No item selected in the project explorer`

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts [hunk 3]
```diff
@@ -28,33 +29,47 @@ export class ProjectPathVariableContribution implements VariableContribution {
...
+                if (!selectionUri.startsWith(rootWorkspaceUri)) {
+                    errorMessage += 'The selection isn\'t under the current workspace root folder.';
+                    await this.messageService.error(errorMessage);
                     return undefined;
                 }
 
-                const relativeSelectionPath = selectionPath.substr(rootWorkspacePath.length);
+                const relativeSelectionPath = selectionUri.substr(rootWorkspaceUri.length);
                 if (!relativeSelectionPath.length) {
+                    errorMessage += 'The selection is on workspace root folder. Select a project.';
```
Comment (sunix) [inline che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: rephrase it `Workspace root folder is selected and is not a project. Please select a project.`

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts [hunk 4]
```diff
@@ -28,33 +29,47 @@ export class ProjectPathVariableContribution implements VariableContribution {
...
 
     async registerVariables(variables: VariableRegistry): Promise<void> {
 
         variables.registerVariable({
             name: 'current.project.path',
             description: 'The path of the project root folder',
             resolve: async () => {
-                const wsRoot = await this.workspaceService.root;
-                if (!wsRoot || !wsRoot.uri) {
+                let errorMessage = 'Get \'current.project.path\' variable error. ';
```
Comment (sunix) [inline che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: rephrase it:
`Can not resolve \'current.project.path\' variable. `

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts (hunk 2 above)
Comment (olexii4) [reply che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: done

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts (hunk 3 above)
Comment (olexii4) [reply che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: Done

File: che-theia-task-extension/src/browser/che-task-variables-contribution.ts (hunk 4 above)
Comment (olexii4) [reply che-theia-task-extension/src/browser/che-task-variables-contribution.ts]: Done

**Review by sunix — APPROVED** [review]
